app/models/ability.rb

```diff
@@ -57,6 +57,14 @@ def initialize(user)
       end
     end
 
+    can :manage_governing_apo, ActiveFedora::Base do |dor_item, new_apo_id|
+      next false unless dor_item.can_manage_item?(user.roles(new_apo_id))
+
+      if dor_item.admin_policy_object
```

Comment:
This if block is repeated from line 48. Can you try to see if we can "call" the cancan block on L48 so we don't have to  copy the code?